src/core/lib/security/transport/client_auth_filter.cc

```diff
@@ -275,22 +275,6 @@ static void auth_start_transport_stream_op_batch(
   call_data* calld = static_cast<call_data*>(elem->call_data);
   channel_data* chand = static_cast<channel_data*>(elem->channel_data);
 
-  if (!batch->cancel_stream) {
-    GPR_ASSERT(batch->payload->context != nullptr);
-    if (batch->payload->context[GRPC_CONTEXT_SECURITY].value == nullptr) {
-      batch->payload->context[GRPC_CONTEXT_SECURITY].value =
```

Comment:
Just want to confirm where is code that batch->payload->context is set to be the same as args->context?

Reply:
It's passed into the filters' `init_call_elem()` here:

https://github.com/grpc/grpc/blob/master/src/core/lib/surface/call.cc#L442

And set in the payload batch here:

https://github.com/grpc/grpc/blob/master/src/core/lib/surface/call.cc#L370

Reply:
Thanks for pointing out. @markdroth

Reply:
@markdroth It seems we may not be able to set the auth_context in init_call_elem() and that may be the cause of the test failures. I'll talk to you offline about this to confirm.